### PR TITLE
Rename Util Random Generates to Prng

### LINKED ISSUE #4054: Rename RandomGenerate to PRNG
### Problem

The protobuf renamed the `RandomGenerateTransaction` to `PrngTransaction`.

### Solution

* Rename `TransactionType.RANDOMGENERATE` to `PRNG`
* Rename `util_random_generate` table to `prng`
* Rename `prng.pseudorandom_bytes` to `prng.prng_bytes`
* Rename `prng.pseudorandom_number` to `prng.prng_number`
* Rename `RandomGenerateTransactionHandler` to `PrngTransactionHandler`
* Rename anything else as appropriate

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Rename Util Random Generates to Prng

**Related issue(s)**:

Fixes #4054

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public class SqlEntityListener implements EntityListener, RecordStreamFileListen
     private final Collection<NftAllowance> nftAllowances;
     private final Collection<NodeStake> nodeStakes;
     private final Collection<NonFeeTransfer> nonFeeTransfers;
-    private final Collection<UtilRandomGenerate> randomGenerates;
+    private final Collection<Prng> pseudoRandomNumbers;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:113]:
```suggestion
    private final Collection<Prng> prngs;
```

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__create_tables.sql [hunk 2]
```diff
@@ -562,12 +562,12 @@ create table if not exists transaction_signature
 );
 comment on table transaction_signature is 'Transaction signatories';
 
--- util_random_generate
-create table if not exists util_random_generate
+-- prng
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__create_tables.sql:565]: Re-sort within file

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.3__index_init.sql [hunk 3]
```diff
@@ -232,9 +232,9 @@ create index if not exists transaction_signature__entity_id
 create index if not exists transaction_signature__timestamp_public_key_prefix
     on transaction_signature (consensus_timestamp desc, public_key_prefix);
 
--- util_random_generate
-alter table util_random_generate
-    add constraint util_random_generate__pk primary key (consensus_timestamp);
+-- prng
+alter table prng
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.3__index_init.sql:236]: Re-sort

File: hedera-mirror-common/src/test/java/com/hedera/mirror/common/domain/DomainBuilder.java [hunk 4]
```diff
@@ -754,11 +754,11 @@ public DomainWrapper<TransactionSignature, TransactionSignature.TransactionSigna
         return new DomainWrapperImpl<>(builder, builder::build);
     }
 
-    public DomainWrapper<UtilRandomGenerate, UtilRandomGenerate.UtilRandomGenerateBuilder> utilRandomGenerate() {
-        var builder = UtilRandomGenerate.builder()
+    public DomainWrapper<Prng, Prng.PrngBuilder> prng() {
```
Comment (steven-sheehy) [inline hedera-mirror-common/src/test/java/com/hedera/mirror/common/domain/DomainBuilder.java:757]: Sort

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java]: Got it, this has been renamed.

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__create_tables.sql (hunk 2 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__create_tables.sql]: Thanks for spotting these, I have fixed its sorting.

File: hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.3__index_init.sql (hunk 3 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.3__index_init.sql]: Got it, fixed.

File: hedera-mirror-common/src/test/java/com/hedera/mirror/common/domain/DomainBuilder.java (hunk 4 above)
Comment (edwin-greene) [reply hedera-mirror-common/src/test/java/com/hedera/mirror/common/domain/DomainBuilder.java]: Resorted.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**